src/Microsoft.VisualStudio.Editors/PropPages/PackagePropPage.vb

```diff
@@ -230,7 +285,113 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
+                SetLicenseRadioButtons(True)
+            End If
+        End Sub
+
+        Private Sub LicenseFileRadioButton_CheckChanged(sender As Object, e As EventArgs) Handles LicenseFileRadioButton.CheckedChanged
+            If (LicenseFileRadioButton.Checked) Then
+                SetLicenseRadioButtons(False)
+            End If
+        End Sub
+
+        Private Sub PackageLicenseExpression_Changed(sender As Object, e As EventArgs) Handles PackageLicenseExpression.TextChanged
+            If (PackageLicenseExpression.Text IsNot "" And Not PackageLicenseExpression.Enabled) Then
+                'The license expression is not selected, and the text was changed while it was disabled
+                'This means there was probably an undo which populated the textbox with text, so give it back control
+                'I don't believe that undo will work with the license file text box because it is not user populated
+                SetLicenseRadioButtons(True)
+            End If
+        End Sub
+
+        'These GotFocus methods are for when the property page is first entered and neither have a value
+        'It would make sense to selected the corresponding radio button when the text box recieves focus
+        Private Sub PackageLicenseExpression_GotFocus(sender As Object, e As EventArgs) Handles PackageLicenseExpression.GotFocus
+            SetLicenseRadioButtons(True)
+        End Sub
+
+        Private Sub LicenseFileNameTextBox_GotFocus(sender As Object, e As EventArgs) Handles LicenseFileNameTextBox.GotFocus
+            SetLicenseRadioButtons(False)
+        End Sub
+
+        Private Sub LicenseBrowseButton_GotFocus(sender As Object, e As EventArgs) Handles LicenseBrowseButton.GotFocus
+            SetLicenseRadioButtons(False)
+        End Sub
+
+        Private Shared Function GetRelativePath(fullPath As String, basePath As String) As String
+            Dim baseUri = New Uri(basePath)
+            Dim fullUri = New Uri(fullPath)
+            Dim relativeUri = baseUri.MakeRelativeUri(fullUri)
+            Return relativeUri.ToString().Replace("/", "\")
+        End Function
+
+        Private Sub AddLicenseItemToProject(fileName As String)
+            Dim unconfiguredProject = GetUnconfiguredProject(ProjectHierarchy)
+            Dim configuredProject As ProjectSystem.ConfiguredProject = ThreadHelper.JoinableTaskFactory.Run(Function()
+                                                                                                                Return unconfiguredProject.GetSuggestedConfiguredProjectAsync
+                                                                                                            End Function)
+            Dim projectSourceItemProvider = configuredProject.Services.ExportProvider.GetExportedValue(Of ProjectSystem.IProjectSourceItemProvider)()
+            Dim correctDirectory = Directory.GetParent(unconfiguredProject.FullPath).ToString
```

Comment:
This is probably just a personal preference thing but if you just need the string, not the full DirectoryInfo, I prefer using Path.GetDirectoryName

Reply:
Good point. I'll change that.